wallet/wallet_test.go

```diff
@@ -0,0 +1,363 @@
+package wallet_test
+
+import (
+	"crypto/sha256"
+	ec "github.com/bsv-blockchain/go-sdk/primitives/ec"
+	"github.com/bsv-blockchain/go-sdk/wallet"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestWallet(t *testing.T) {
+	// Create test data
+	sampleData := []byte{3, 1, 4, 1, 5, 9}
+
+	// Generate keys
+	userKey, err := ec.NewPrivateKey()
+	assert.NoError(t, err)
+	counterpartyKey, err := ec.NewPrivateKey()
+	assert.NoError(t, err)
+
+	// Create wallets with proper initialization
+	userWallet := wallet.NewWallet(userKey)
+	counterpartyWallet := wallet.NewWallet(counterpartyKey)
+
+	// Define protocol and key ID
+	protocol := wallet.WalletProtocol{
+		SecurityLevel: wallet.SecurityLevelEveryAppAndCounterparty,
+		Protocol:      "tests",
+	}
+	keyID := "4"
+
+	// Encrypt message
+	encryptArgs := &wallet.WalletEncryptArgs{
+		WalletEncryptionArgs: wallet.WalletEncryptionArgs{
+			ProtocolID: protocol,
+			KeyID:      keyID,
+			Counterparty: wallet.WalletCounterparty{
+				Type:         wallet.CounterpartyTypeOther,
+				Counterparty: counterpartyKey.PubKey(),
+			},
+		},
+		Plaintext: sampleData,
+	}
+
+	encryptResult, err := userWallet.Encrypt(encryptArgs)
+	assert.NoError(t, err)
+	assert.NotEqual(t, sampleData, encryptResult.Ciphertext)
+
+	// Decrypt message
+	decryptArgs := &wallet.WalletDecryptArgs{
+		WalletEncryptionArgs: wallet.WalletEncryptionArgs{
+			ProtocolID: protocol,
+			KeyID:      "4",
+			Counterparty: wallet.WalletCounterparty{
+				Type:         wallet.CounterpartyTypeOther,
+				Counterparty: userKey.PubKey(),
+			},
+		},
+		Ciphertext: encryptResult.Ciphertext,
+	}
+
+	decryptResult, err := counterpartyWallet.Decrypt(decryptArgs)
+	assert.NoError(t, err)
+	assert.Equal(t, sampleData, decryptResult.Plaintext)
```

Comment:
maybe wrap it also into t.Run, so in case of failure it would be clearer that the test of encryption has failed.